packages/backend/server/src/plugins/copilot/context/service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaClient } from '@prisma/client';
+
+import {
+  Cache,
+  CopilotInvalidContext,
+  CopilotSessionNotFound,
+} from '../../../base';
+import { ContextSession } from './session';
+import { ContextConfig, ContextConfigSchema } from './types';
+
+const CONTEXT_SESSION_KEY = 'context-session';
+
+@Injectable()
+export class CopilotContextService {
+  constructor(
+    private readonly cache: Cache,
+    private readonly db: PrismaClient
+  ) {}
+
+  private async saveConfig(
+    contextId: string,
+    config: ContextConfig,
+    refreshCache = false
+  ): Promise<void> {
+    if (!refreshCache) {
+      await this.db.aiContext.update({
+        where: { id: contextId },
+        data: { config },
+      });
+    }
+    await this.cache.set(`${CONTEXT_SESSION_KEY}:${contextId}`, config);
+  }
+
+  private async getCachedSession(
+    contextId: string
+  ): Promise<ContextSession | undefined> {
+    const cachedSession = await this.cache.get(
+      `${CONTEXT_SESSION_KEY}:${contextId}`
+    );
+    if (cachedSession) {
+      const config = ContextConfigSchema.safeParse(cachedSession);
+      if (config.success) {
+        return new ContextSession(
+          contextId,
+          config.data,
+          this.saveConfig.bind(this, contextId)
+        );
+      }
+    }
+    return undefined;
+  }
+
+  private async cacheSession(
+    contextId: string,
+    config: ContextConfig
+  ): Promise<ContextSession> {
+    const dispatcher = this.saveConfig.bind(this, contextId);
```

Comment:
What's the purpose of the `bind` here? Why not just `await this.saveConfig(contextId, true)`?